lib/serviceClient.ts

```diff
@@ -136,6 +141,11 @@ export class ServiceClient {
     } catch (error) {
       return Promise.reject(error);
     }
+
+    if (operationSpec) {
```

Comment:
I am little confused over here. Wouldn't the options have the http method in it. I believe RequestPrepareOptions had that. Why would someone provide the http method in the operationSpec then?

Do you want to start with a completely new method named sendRequestWithOperationSpec() rather than touching this one?

Reply:
Thus the generic sendRequest can stay as-is and will be simple to use for customers and the other flavor will be a dedicated one for operation spec. It will save us from bunch of if-else clauses (which always turn out to be potential disasters, once things get complex).

Reply:
`options` (aka `httpRequest`) may have the HTTP method on it. We really have two scenarios here:
1. sendRequest() is invoked by AutoRest-generated code, and
2. sendRequest() is invoked by customer code that wasn't generated by AutoRest.
My plan is for all of our AutoRest-generated code to just pass the arguments provided to the AutoRest-generated method plus an OperationSpec to ServiceClient.sendRequest() and for sendRequest() to create the HttpRequest and to send it off. However, I can't really do that yet because I'm trying to take this in small steps.
It might make a lot of sense to create an additional method called `sendOperationRequest()` separate from `sendRequest()`, but I'm planning on holding off on that until I have pushed all of the logic in our generated methods into the ServiceClient.sendRequest() method.